src/compose/ComposeBox.js

```diff
@@ -207,26 +289,35 @@ export default function ComposeBox(props: Props): Node {
     return focusState.either;
   }, [isEditing, narrow, focusState.either]);
 
+  /**
+   * Inserts text at the message input's cursor position.
+   *
+   * Selected text is not replaced: selection-start is ignored, and
+   * selection-end is taken to be the cursor position.
+   *
+   * If padWithEmptyLines is true, adds an empty line before (unless cursor
+   * is at 0) and after `text`.
+   */
+  // An earlier version of this function *would* replace selected text,
+  // which could be upsetting if you didn't want to lose the selected text.
   const insertMessageTextAtCursorPosition = useCallback(
-    (text: string) => {
-      setMessageInputValue(
-        state =>
-          state.value.slice(0, state.selection.start)
-          + text
-          + state.value.slice(state.selection.end, state.value.length),
-      );
-    },
-    [setMessageInputValue],
-  );
+    (text: string, padWithEmptyLines = false) => {
+      setMessageInputValue(state => {
+        const cursorPosition = state.selection.end;
 
-  const insertVideoCallLinkAtCursorPosition = useCallback(
-    (url: string) => {
-      const linkMessage = _('Click to join video call');
-      const linkText = `[${linkMessage}](${url})`;
+        const paddingStart = padWithEmptyLines && cursorPosition !== 0 ? '\n\n' : '';
+        const paddingEnd = padWithEmptyLines ? '\n\n' : '';
```

Comment:
Why two newlines instead of one? See [discussion](https://chat.zulip.org/#narrow/stream/101-design/topic/Quote-and-reply.20newlines/near/1465208):

> I think the extra breathing room could be nice

Reply:
That discussion led to https://github.com/zulip/zulip/issues/23608 for doing this in the web app too.